plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -1919,6 +1924,410 @@ public void testQueryTableFunction()
+                                        .put("array_string_field", ImmutableList.<String>builder()
+                                                        .addAll(Stream.of("trino", "the", "lean", "machine-ohs")::iterator)
+                                                        .build())
+                                        .put("string_field", "sample")
+                                        .put("int_field", 2)
+                                        .put("object_field_2", ImmutableMap.<String, Object>builder()
+                                                        .put("array_string_field", ImmutableList.<String>builder()
+                                                                .addAll(Stream.of("trino", "the", "lean", "machine-ohs")::iterator)
+                                                                .build())
+                                                        .put("string_field2", "sample")
+                                                        .put("int_field", 33)
+                                                        .put("object_field_3", ImmutableMap.<String, Object>builder()
+                                                                        .put("array_string_field", ImmutableList.<String>builder()
+                                                                                .addAll(Stream.of("trino", "the", "lean", "machine-ohs")::iterator)
+                                                                                .build())
+                                                                        .put("string_field3", "some value")
+                                                                        .put("int_field3", 55)
+                                                                .buildOrThrow())
+                                                .buildOrThrow())
+                                .buildOrThrow())
+                        .buildOrThrow());
+
+        HashMap<String, Object> innerRecord = new HashMap<>();
+        innerRecord.put("object_field_inner", null);
+        index(tableName, ImmutableMap.<String, Object>builder()
+                .put("long_field", 11122233L)
+                .put("object_field_outer", innerRecord)
+                .buildOrThrow());
+
+        assertQuery("select id_field, object_field.object_field_2.object_field_3.string_field3 from " + tableName + " where  object_field_outer.int_field_outer=44", "VALUES ('564e6982-88ee-4498-aa98-df9e3f6b6109', 'some value')");
+        assertQuery("select object_field_outer.int_field_outer from " + tableName + " where  object_field_outer.int_field_outer=44", "VALUES 44");
+        assertQuery("select long_field, id_field, object_field.object_field_2.object_field_3.string_field3, object_field_outer.object_field_inner.int_field_inner from " + tableName + " where  long_field=11122233", "VALUES (11122233, NULL, NULL, NULL)");
```

Comment:
Do we also need `.isFullyPushedDown()` here?

Reply:
`.isFullyPushedDown()` compares the result with query being pushed down and not pushed down. In later case, query will fails as I have not defined a mapping for Index having an array field in nested field. It is basically the failing scenario that we are trying to solve with allowing dereferences to be pushed down.